### PR TITLE
feature: add '--- curl_options' and '--- curl_protocol' section

### PR DESCRIPTION
**Pull request by woodgear**
feature: add '--- curl_options' and '--- curl_protocol' section which could combine with '--- http2' to construct a request by curl.

### CODE REVIEW

**Review by zhuizhuhaomeng — APPROVED** [review]

**Comment by xiaocang** [comment]
Hi @woodgear, documentation is required here

**Comment by xiaocang** [comment]
The Test::Nginx document is in pod format (check https://perldoc.perl.org/perlpod for help), and the documentation for these two new options can be just added to the current .pm file like other options

**Comment by woodgear** [comment]
update to add doc。 not sure I'm doing it right.

File: lib/Test/Nginx/Socket.pm [hunk 1]
```diff
@@ -2993,6 +3003,14 @@ C<--- raw_request>.
 
 See also the L<TEST_NGINX_USE_HTTP2> system environment for the "http2" test mode.
 
+=head2 curl_options
+
+add curl options when use 'curl' to send request.
+
+=head2 curl_protocol
+
+set protocol such as http/https when use 'curl' to send request.
```
Comment (xiaocang) [inline lib/Test/Nginx/Socket.pm:3012]: `Set protocol (such as http/https) when sending request using 'curl'` may be better here

File: lib/Test/Nginx/Socket.pm [hunk 2]
```diff
@@ -2993,6 +3003,14 @@ C<--- raw_request>.
 
 See also the L<TEST_NGINX_USE_HTTP2> system environment for the "http2" test mode.
 
+=head2 curl_options
+
+add curl options when use 'curl' to send request.
```
[the remark this answers is not in the document]
Comment (xiaocang) [reply lib/Test/Nginx/Socket.pm:3008]: And, you can give a useful example here

**Review by xiaocang — CHANGES_REQUESTED** [review]

**Comment by woodgear** [comment]
update

File: lib/Test/Nginx/Socket.pm [hunk 3]
```diff
@@ -2993,6 +3003,23 @@ C<--- raw_request>.
...
+
+Set protocol (such as http/https) when sending request using 'curl'.
+
+=head2 curl_options
+
+Add extra command line options when using 'curl' to send request.
+
+Below example will send https request via curl:
+
+--- http2
```
Comment (xiaocang) [inline lib/Test/Nginx/Socket.pm:3016]: The sample code block should start with the tab

**Review by xiaocang — CHANGES_REQUESTED** [review]

File: lib/Test/Nginx/Socket.pm (hunk 3 above)
Comment (woodgear) [reply lib/Test/Nginx/Socket.pm]: update

**Review by xiaocang — APPROVED** [review]
LGTM

**Comment by xiaocang** [comment]
Merged with minor tweaks, thanks for the contribution!

``` patch
diff --git a/lib/Test/Nginx/Socket.pm b/lib/Test/Nginx/Socket.pm
index 6901192..398fbd8 100644
--- a/lib/Test/Nginx/Socket.pm
+++ b/lib/Test/Nginx/Socket.pm
@@ -3011,7 +3011,7 @@ Set protocol (such as http/https) when sending request using 'curl'.
 
 Add extra command line options when using 'curl' to send request.
 
-Below example will send https request via curl:
+Below is an example for sending an insecure https request using 'curl':
 
     --- http2
     --- curl_options: -k
```

**Comment by woodgear** [comment]
cool  ^_^.

**Pull request merged**